beater/handlers.go

```diff
@@ -133,6 +137,55 @@ func isAuthorized(req *http.Request, secretToken string) bool {
 	return subtle.ConstantTimeCompare([]byte(parts[1]), []byte(secretToken)) == 1
 }
 
+func corsHandler(allowedOrigins []string, h http.Handler) http.Handler {
+
+	var isAllowed = func(origin string) bool {
+		for _, allowed := range allowedOrigins {
+			if origin == allowed {
+				return true
+			}
+		}
+		return len(allowedOrigins) == 1 && allowedOrigins[0] == "*"
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+		// origin header is always set by the browser
+		origin := r.Header.Get("Origin")
+		validOrigin := isAllowed(origin)
+
+		if r.Method == "OPTIONS" {
+
+			// setting the ACAO header is the way to tell the browser to go ahead with the request
+			if validOrigin {
+				// do not set the configured origin(s), echo the received origin instead
+				w.Header().Set("Access-Control-Allow-Origin", origin)
+			}
+
+			// tell browsers to cache response requestHeaders for up to 1 hour (browsers might ignore this)
+			w.Header().Set("Access-Control-Max-Age", "3600")
+			// origin must be part of the cache key so that we can handle multiple allowed origins
+			w.Header().Set("Vary", "Origin")
+
+			// required if Access-Control-Request-Method and Access-Control-Request-Headers are in the requestHeaders
+			w.Header().Set("Access-Control-Allow-Methods", supportedMethods)
+			w.Header().Set("Access-Control-Allow-Headers", supportedHeaders)
```

Comment:
I think we need these on the POST as well. @jahtalab can you confirm?

Reply:
@roncohen , We only need Access-Control-Allow-Origin on the POST request, As I summarized in another comment